src/sif/truckcost.cc

```diff
@@ -484,7 +485,9 @@ Cost TruckCost::EdgeCost(const baldr::DirectedEdge* edge,
                                          &flow_sources, time_info.seconds_from_now)
                         : fixed_speed_;
 
-  auto final_speed = std::min(edge_speed, top_speed_);
+  auto final_speed = std::min(std::min(edge_speed, edge->truck_speed() ? edge->truck_speed()
+                                                                       : kMaxAssumedTruckSpeed),
+                              top_speed_);
```

Comment:
Basically the same as in `shortcutbuilder`, but here we need to take into account the `top_speed` as well: we want to let the user specifiy top speeds higher than our new `kMaxAssumedTruckSpeed`, at least up to `kMaxSpeedKph`.